src/main.cc

```diff
@@ -61,19 +118,17 @@ int main(int argc, char** argv) {
     glfwSwapBuffers(window);
 
     // Create a nanogui screen
-    auto* screen = new nanogui::Screen();
-    screen->initialize(window, true);
-    std::cout << "Created screen" << std::endl;
-
-    screen->setVisible(true);
-    screen->performLayout();
-
+    screen = new nanogui::Screen();
     std::cout << "Completed Initialization" << std::endl;
 
-    SmokeSimulator sim(shader_path, screen);
+    sim = new SmokeSimulator(shader_path, screen, window);
```

Comment:
I think stack local variable would be sufficient here? The scope of this would not exceed main